specs/scenarios/trust-store-and-trust-policy.md

```diff
@@ -0,0 +1,212 @@
+| Wabbit Networks | Wabbit Networks Batch private key | `batch.wabbitnetworkscr.io` |
+| ACME Rockets | ACME Rockets Map/Reduce private key | `mr.acmecr.io` |
+
+## Scenario: Trust for Multi-Tenant Services
+With the background from the above, this section will describe the trust scenario for multi-tenant services.
+
+### Delivering the Trust Store and Trust Policy for the Plaform Layer
+A Kubernetes node is created (either when a new cluster is created or an existing one is expanded) from an existing VM image. For Kubernetes management plane to be able to pull the platform containers, it needs to have the trust certificates already available on the node. Therefore, the VM image needs to deliver the trust certificates for the *platform layer* as part of the VM image itself. 
+
+In our case, the VM image should already have the *trust store* created on the VM file system and have the certificates for the *platform layer* available on the file system. Here is how the *trust store* should look like when the node is spun up.
+
+```
+$XDG_CONFIG_HOME/notation/trust-store
+    /x509
+        /ca
+            /wabbit-networks
+                wabbit-networks-kubernetes.crt
+        /tsa
+            /publicly-trusted-tsa
+                tsa-cert1.pem
+```
+However, just the *trust store* is not enough for Notation to validate the containers used for the *platform layer*. A *trust policy* needs to be established for the *platform layer* containers. Here is how the trust policy needs to look for a brand new node in a Kubernetes cluster based on the above example.
+
+```
+{
+    "version": "1.0",
+    "trustPolicies": [
+        {
+            "name": "wabbit-networks-kubernetes",
+            "registryScopes": [ "k8s.wabbitnetworkscr.io/k8s" ],          
+            "signatureVerification": {          
+              "level" : "strict" 
+            },
+            "trustStores": ["ca:wabbit-networks"], 
+            "trustedIdentities": [
+              "x509.subject: C=US, ST=WA, L=Seattle, O=wabbitnetworks.com, OU=KubernetesTeam, CN=KuberenetesTeamSecureBuilder"
+            ]
+        }
+    ]
+}
+```
+
+The above *trust store* and *trust policy* are the initial state for each Kubernetes node in the cluster once the node is spun up. Once again, those are baked into the VM image that is used to start the node.
+
+### Adding Trust Certificates to the Trust Store and Trust Policy for the Service Layer
+Once the node is started the Wabbit Networks Batch service needs to be deployed on the node. For that to work, the *trust store* needs to be updated with the Wabbit Networks Batch certificates and a trust policy needs to be added. Here is how the *trust store* should look like:
+
+```
+$XDG_CONFIG_HOME/notation/trust-store
+    /x509
+        /ca
+            /wabbit-networks
+                wabbit-networks-kubernetes.crt
+                wabbit-networks-batch.crt
+        /tsa
```

Comment:
If `wabbit-networks-kubernetes.crt` and `wabbit-networks-batch.crt are derived from same root, you can directly add that root.

Reply:
That is correct. Though, I normally question the security to trust broad roots.